app.rb

```diff
@@ -0,0 +1,122 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]:'
+    input = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+    case input
+    when 1
+      create_student(age, name)
+    when 2
+      create_teacher(age, name)
+    end
+    puts 'Person created successfully!'
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+
+    parent_permission = gets.chomp.upcase == 'Y'
+    @people << Student.new(name, age, parent_permission)
+
+    puts 'Student created successfully'
+    sleep 0.75
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people << Teacher.new(specialization, age, name)
+
+    puts 'Teacher created successfully'
+    sleep 0.75
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
```

Comment:
- [ ] You are yet to implement the change requested by the previous reviewer. The sole responsibility of the `create_book` method should be to save the data in line with the Single-responsibility rule of SOLID. The process of printing information and getting user inputs should be handled by another method/class. Kindly refactor accordingly.

  **SUGGESTION:** You can extract lines 82 to 86 into a different method and have that method return your `title` and `author` variables in an array (see first image) then in `create_book`, you can destructure your variables from the method (see the second image)

  ![somdo-details](https://user-images.githubusercontent.com/36542986/151937268-6aa8928c-11ea-4050-85ad-11a726776201.jpg) ![somdo-use-var](https://user-images.githubusercontent.com/36542986/151937273-14abc64c-8ae9-4a2a-b5af-a1230a7c13a5.jpg)

Reply:
Created book_details as stated by the previous reviewer